src/parser/mod.rs

```diff
@@ -8100,6 +8115,17 @@ impl<'a> Parser<'a> {
             );
         };
 
+        self.parse_remaining_set_exprs(expr, precedence)
+    }
+
+    /// Parse any extra set expressions that may be present in a query body
+    ///
```

Comment:
I also pushed a commit to split up this function in e2ec2a6 to get the deeply nested tests to pass